test/langtools/tools/javac/annos_not_copied_anonymous_ctr/AnnosNotCopiedToAnonymousCtrTest.java

```diff
@@ -0,0 +1,123 @@
+                        method.attributes,
+                        "Annotations hasn't been propagated",
+                        new AnnoData(Attribute.RuntimeVisibleAnnotations, "LAnnosNotCopiedToAnonymousCtrTest$VisibleCtrAnnotation;", -1),
+                        new AnnoData(Attribute.RuntimeInvisibleAnnotations, "LAnnosNotCopiedToAnonymousCtrTest$InvisibleCtrAnnotation;", -1),
+                        new AnnoData(Attribute.RuntimeVisibleParameterAnnotations, "LAnnosNotCopiedToAnonymousCtrTest$VisibleParamAnnotation;", 1),
+                        new AnnoData(Attribute.RuntimeInvisibleParameterAnnotations, "LAnnosNotCopiedToAnonymousCtrTest$InvisibleParamAnnotation;", 2) );
+            }
+        }
+    }
+
+    void checkForAttr(ClassFile classFile, Attributes attrs, String errorMsg, AnnoData... attrAndParamPos)
+            throws Throwable {
+        for (AnnoData attrAndPos : attrAndParamPos) {
+            Assert.checkNonNull(attrs.get(attrAndPos.attributeName()), errorMsg);
+            boolean isParamAnno = attrs.get(attrAndPos.attributeName()) instanceof RuntimeParameterAnnotations_attribute;
+            if (isParamAnno) {
+                RuntimeParameterAnnotations_attribute paramAnnotation =
+                        (RuntimeParameterAnnotations_attribute)attrs.get(attrAndPos.attributeName());
+                for (int i = 0; i < paramAnnotation.parameter_annotations.length; i++) {
+                    Annotation[] annos = paramAnnotation.parameter_annotations[i];
+                    if (i != attrAndPos.positionOfAnnotatedParam()) {
+                        Assert.check(annos.length == 0);
+                    } else {
+                        Assert.check(annos.length == 1);
+                        Assert.check(classFile.constant_pool.getUTF8Value(annos[0].type_index).equals(attrAndPos.annoName()));
+                    }
+                }
+            } else {
+                RuntimeAnnotations_attribute ctrAnnotation = (RuntimeAnnotations_attribute)attrs.get(attrAndPos.attributeName());
+                Assert.check(ctrAnnotation.annotations.length == 1);
+                Assert.check(classFile.constant_pool.getUTF8Value(ctrAnnotation.annotations[0].type_index).equals(attrAndPos.annoName()));
+            }
+        }
+    }
+
+    @Target(value = {ElementType.PARAMETER})
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface VisibleParamAnnotation {}
+
+    @Target(value = {ElementType.PARAMETER})
+    @interface InvisibleParamAnnotation {}
+
+    @Target(value = {ElementType.CONSTRUCTOR})
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface VisibleCtrAnnotation {}
+
+    @Target(value = {ElementType.CONSTRUCTOR})
+    @interface InvisibleCtrAnnotation {}
+
+    public class Test {
+        @VisibleCtrAnnotation
+        @InvisibleCtrAnnotation
+        public Test(String firstParam, @VisibleParamAnnotation String secondParam, @InvisibleParamAnnotation String thirdParam) {}
```

Comment:
How about type annotations of the parameter types? Should these be copied? (And tested?)

Reply:
yep I should add a test for type annotations, good point

Reply:
I have added support for type annotations, thanks for the comments